test/functional/utils/test-rate-limit.js

```diff
@@ -65,4 +65,22 @@ describe('rate-limit', () => {
     expect(callback).to.be.calledOnce;
     expect(callback).to.be.calledWith(7, 'another param');
   });
+
+  it('is re-entrant', () => {
+    let calls = 0;
+    let rateLimitedCallback;
+    function fn(call) {
+      if (calls++ < 1) {
+        rateLimitedCallback(calls);
```

Comment:
can you explain more about what this test trying to say?
and also, which line of change made this function re-entrant?